apps/docs/pages/docs/api-reference/fields/richtext.mdx

```diff
@@ -0,0 +1,557 @@
+        },
+      },
+    },
+    defaultProps: {
+      body: "<p>Hello, world</p>",
+    },
+    render: ({ body }) => body,
+  }}
+>
+  <Puck.Preview />
+</ConfigPreview>
+
+#### Included extensions
+
+- [`Blockquote`](https://tiptap.dev/docs/editor/extensions/nodes/blockquote)
+- [`Bold`](https://tiptap.dev/docs/editor/extensions/marks/bold)
+- [`BulletList`](https://tiptap.dev/docs/editor/extensions/nodes/bullet-list)
+- [`Code`](https://tiptap.dev/docs/editor/extensions/marks/code)
+- [`CodeBlock`](https://tiptap.dev/docs/editor/extensions/nodes/code-block)
+- [`Document`](https://tiptap.dev/docs/editor/extensions/nodes/document)
+- [`HardBreak`](https://tiptap.dev/docs/editor/extensions/nodes/hard-break)
+- [`Heading`](https://tiptap.dev/docs/editor/extensions/nodes/heading)
+- [`HorizontalRule`](https://tiptap.dev/docs/editor/extensions/nodes/horizontal-rule)
+- [`Italic`](https://tiptap.dev/docs/editor/extensions/marks/italic)
+- [`ListItem`](https://tiptap.dev/docs/editor/extensions/nodes/list-item)
+- [`ListKeymap`](https://tiptap.dev/docs/editor/extensions/functionality/listkeymap)
+- [`Link`](https://tiptap.dev/docs/editor/extensions/marks/link)
+- [`OrderedList`](https://tiptap.dev/docs/editor/extensions/nodes/ordered-list)
+- [`Paragraph`](https://tiptap.dev/docs/editor/extensions/nodes/paragraph)
+- [`Strike`](https://tiptap.dev/docs/editor/extensions/marks/strike)
+- [`Text`](https://tiptap.dev/docs/editor/extensions/nodes/text)
+- [`TextAlign`](https://tiptap.dev/docs/editor/extensions/functionality/textalign)
+- [`Underline`](https://tiptap.dev/docs/editor/extensions/marks/underline)
+
+### `renderMenu(props)`
+
+Render a custom menu for the field. Use with the [`<RichTextMenu>`](/docs/api-reference/components-rich-text-menu) component to arrange existing controls, or introduce new ones.
+
+Receives [Menu Render Props](#menu-render-props).
+
+```tsx {1, 9-23} copy showLineNumbers
+import { RichTextMenu } from "@measured/puck";
+
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          renderMenu: ({ children }) => (
+            <RichTextMenu>
+              {/* Render default menu */}
+              {children}
+
+              <RichTextMenu.Group>
+                {/* Render a Blockquote control */}
+                <RichTextMenu.Blockquote />
+
+                {/* Render a custom control */}
+                <RichTextMenu.Control {/* ... */} />
+
+              </RichTextMenu.Group>
+            </RichTextMenu>
+          ),
+        },
+      },
+      // ...
+    },
+  },
+};
+```
+
+<ConfigPreview
+  label="Example"
+  componentConfig={{
+    fields: {
+      body: {
+        type: "richtext",
+        renderMenu: ({ children }) => (
+          <RichTextMenu>
+            {children}
+            <RichTextMenu.Group>
+              <RichTextMenu.Control icon={<Quote />} />
+              <RichTextMenu.Control icon={<Highlighter />} />
```

Comment:
Should highlighter do anything or is it just indicative of adding a control?

Reply:
It was just meant to be indicative. I've added an `alert` on click.

Reply:
It was just meant to be indicative. I've added an `alert` on click.